### DISCUSSION TITLE
Pointer Gamut Coverage Reconciliation

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1164] [Q&A] [answered]
I've compared Colour's output for Pointer Gamut coverage using the Monte Carlo method to what looks like a vigorous study carried out by Simon Baker at TFT Central:   [https://tftcentral.co.uk/articles/pointers_gamut](https://tftcentral.co.uk/articles/pointers_gamut)

Some of the deltas are concerning.  Of course we have to check if it's user (me!) error first.  Here's an example of my calculation of Adobe RGB (1998):

```
from colour.models import RGB_COLOURSPACE_ADOBE_RGB1998 as A98
prng = np.random.RandomState(2)
colour.RGB_colourspace_pointer_gamut_coverage_MonteCarlo(A98, 10e3, random_state=prng)

95.13590844062946
```

Here is the comparison between Colour and Baker:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/jharmon/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/jharmon/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Color   Space | COLOUR Pointer Coverage | BAKER Pointer Coverage | Delta
-- | -- | -- | --
sRGB | 82.07% | 69.40% | 12.67%
Adobe RGB (1998) | 95.14% | 86.20% | 8.96%
Adobe Wide Gamut RGB | 95.37% | 99.10% | -3.73%
DCI-P3 | 88.30% | 86.90% | 1.40%
NTSC (1953) | 98.15% | 89.50% | 8.65%
ProPhoto RGB | 96.06% | 100% | -3.94%
Rec.2020 | 99.93% | 99.90% | 0.03%
ACES RGB | 100% | 100% | 0%

</body>

</html>

I welcome anyone's feedback on this!  Unfortunately, Baker doesn't articulate his method of calculating the volume.  He may be doing it in 2D, since he's so focused on chromaticities.

- J

**Comment by thomasmansencal** [comment, marked as answer]
Yes it is hard to say how his computations were done, we do them in 3D, I would recommend that you adapt the colourspaces to the Pointer's Illuminant `SC`.

**Reply by blue-j** [reply]
SIgh.  I cannot figure out how to adapt an entire RGB colorspace to a different white point.  In reviewing some of my code over here I see also I did not intelligently adapt XYZ values to a different white point either.  can you point me in the right direction?  i'm not lazy and value your time very much!  - J

**Reply by tjdcs** [reply]
It might be helpful to read SMPTE RP177, which can basically help you compute a transformation matrix from one RGB system to another (including balancing for a different whitepoint) this is basically the math behind derived NPMs in RGBColourspace.

But strictly speaking, it is not adaptation which is a physiological "white balancing" in the retina. This is best done using a "chromatic adaptation model" of which there are several in colour. CAT16 is exceedingly popular.

**Reply by blue-j** [reply]
i can do all of that for an array of values, but for a colorspace to a different white point... no entiendo the exact syntax to be used.  I cant crack it

**Reply by blue-j** [reply]
well, i think i figured out how to adapt an XYZ value to a different white point?  an example going from D50 -> D65:

XYZ = np.array([0.20654008, 0.12197225, 0.05136952])
XYZ_w = colour.TVS_ILLUMINANTS['CIE 1931 2 Degree Standard Observer']['D65']
XYZ_wr = colour.TVS_ILLUMINANTS['CIE 1931 2 Degree Standard Observer']['D50']
transform = "Bradford"
XYZ_2 = colour.adaptation.chromatic_adaptation_VonKries(XYZ, XYZ_w, XYZ_wr, transform)

print(XYZ_2)

**Reply by thomasmansencal** [reply]
There is a dedicated method to adapt a RGB colourspace: https://colour.readthedocs.io/en/develop/generated/colour.RGB_Colourspace.html#colour.RGB_Colourspace.chromatically_adapt

Your above code looks correct by the way!

**Comment by blue-j** [comment]
```
import numpy as np
import colour
import warnings

# SUPPRESS WARNINGS
warnings.filterwarnings("ignore")

prng = np.random.RandomState(2)

# chromaticity coordinates for SC light source white point
w_t = np.array([0.31005673, 0.3161457])

for colorspaceid in colour.RGB_COLOURSPACES.keys():
    rd = np.array(colour.RGB_COLOURSPACES[colorspaceid].primaries[0])
    gr = np.array(colour.RGB_COLOURSPACES[colorspaceid].primaries[1])
    bl = np.array(colour.RGB_COLOURSPACES[colorspaceid].primaries[2])
    p = np.array([rd, gr, bl])
    w_r = colour.RGB_COLOURSPACES[colorspaceid].whitepoint
    w_r_name = str(colour.RGB_COLOURSPACES[colorspaceid].whitepoint_name)

    colorspace = colour.RGB_Colourspace(colorspaceid, p, w_t, w_r_name)

    rgb_adapted = colorspace.chromatically_adapt(w_r, "SC", "CAT16")

    pointer_percentage = colour.RGB_colourspace_pointer_gamut_coverage_MonteCarlo(
        rgb_adapted, 10e3, random_state=prng
    )
    print(colorspaceid, ": ", pointer_percentage)
```



Seems to work fine.  Does it look ok?

Thanks both of you for your kind support and mentorship!

- J

**Comment by blue-j** [comment]
there are some values that are a little puzzling:

DCDM XYZ :  9.334248455730954
Max RGB :  14.07103825136612
Xtreme RGB :  2.8275862068965516

But these spaces are not of much concern and can be ignored for our purposes here, just thought I'd report.

- J
